activestorage/lib/active_storage/attached/changes/create_many.rb

```diff
@@ -25,6 +25,14 @@ def upload
     def save
       assign_associated_attachments
       reset_associated_blobs
+
+      unless key.blank?
```

Comment:
Instead of moving blobs after saving them, could we modify their key when we instantiate them based on the configured path? I think that way we could save the additional `move_to!` logic completely.

In general, implicitly moving records on every save sounds like a bit expensive.